docs/source/design.rst

```diff
@@ -1,7 +1,151 @@
-.. _internals:
+.. _design:
 
-*********************
-Ibis design internals
-*********************
+Design
+======
 
-More to come here.
+Primary Goals
+*************
+
+#. Type safety
+#. Expressiveness
+#. Composability
+#. Familiarty
+
+Flow of execution
+-----------------
+
+#. User writes expression
+#. Each method or function call builds a new expression
+#. Expressions are type checked as you create them
+#. Some expressions have some optimizations that happen as the user builds them
+   (EXAMPLE)
+#. Some optimizations happen at compile time? (EXAMPLE)
+#. Expressions are compiled
+#. The SQL string that generated by the compiler is sent to the database and
```

Comment:
this is true for SQL but not for other backends where the execution can happen locally / remote (e.g. pandas / spark / file based)

Reply:
Added a note